app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationService.kt

```diff
@@ -0,0 +1,791 @@
+                    R.string.notification_report_name,
+                    R.string.notification_report_description,
+                ),
+            )
+            // TODO enumerate all keys of Notification.Type and check if one is missing here?
+
+            val channelGroup = NotificationChannelGroup(account.identifier, account.fullName)
+            notificationManager.createNotificationChannelGroup(channelGroup)
+
+            val channels: MutableList<NotificationChannel> = ArrayList(12)
+            channelData.forEach {
+                val name = context.getString(it.name)
+                val description = context.getString(it.description)
+                val importance = NotificationManager.IMPORTANCE_DEFAULT
+                val channel = NotificationChannel(it.id, name, importance)
+
+                channel.description = description
+                channel.enableLights(true)
+                channel.lightColor = -0xd46f27
+                channel.enableVibration(true)
+                channel.setShowBadge(true)
+                channel.group = account.identifier
+                channels.add(channel)
+            }
+
+            notificationManager.createNotificationChannels(channels)
+        }
+    }
+
+    fun deleteNotificationChannelsForAccount(account: AccountEntity) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            notificationManager.deleteNotificationChannelGroup(account.identifier)
+        }
+    }
+
+    fun enablePullNotifications(context: Context) {
+        val workManager = WorkManager.getInstance(context)
+        workManager.cancelAllWorkByTag(NOTIFICATION_PULL_TAG)
+
+        // Periodic work requests are supposed to start running soon after being enqueued. In
+        // practice that may not be soon enough, so create and enqueue an expedited one-time
+        // request to get new notifications immediately.
+        val fetchNotifications: WorkRequest = OneTimeWorkRequest.Builder(NotificationWorker::class.java)
+            .setExpedited(OutOfQuotaPolicy.DROP_WORK_REQUEST)
+            .setConstraints(Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build())
+            .build()
+        workManager.enqueue(fetchNotifications)
+
+        val workRequest: WorkRequest = PeriodicWorkRequest.Builder(
+            NotificationWorker::class.java,
+            PeriodicWorkRequest.MIN_PERIODIC_INTERVAL_MILLIS, TimeUnit.MILLISECONDS,
+            PeriodicWorkRequest.MIN_PERIODIC_FLEX_MILLIS, TimeUnit.MILLISECONDS
+        )
+            .addTag(NOTIFICATION_PULL_TAG)
+            .setConstraints(Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build())
+            .setInitialDelay(5, TimeUnit.MINUTES)
+            .build()
+
+        workManager.enqueue(workRequest)
+
+        Log.d(TAG, "Enabled pull checks with " + PeriodicWorkRequest.MIN_PERIODIC_INTERVAL_MILLIS + "ms interval.")
+    }
+
+    fun disablePullNotifications(context: Context) {
```

Comment:
No need for the `context` parameter, there is already a `Context` injected

Reply:
That is a bit of a question still: Does it matter which context is used (somewhere)?

90% of locations in the old code worked directly or indirectly with application context. But some did and do not.

Reply:
It can matter, yes, for example when taking theme attributes from a context, then you need to use the Activity Context as each Activity can have a different theme. But here it shouldn't 🤔

Reply:
All removed.